internal/integrations/github/github_test.go

```diff
@@ -0,0 +1,52 @@
+package github
+
+import (
+	"io/ioutil"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"gopkg.in/h2non/gock.v1"
+)
+
+func TestGetTravisCIBuildURLForRef_Success(t *testing.T) {
+	defer gock.Off()
+
+	mockResponse, _ := ioutil.ReadFile("testdata/check_runs_response_success.json")
+	gock.New("https://api.github.com").
+		Get("/repos/Wattpad/highlander/commits/2c76895cdb9f3ff5100ecf93a7a6c6747aaeda8c/check-runs").
+		Reply(200).
+		JSON(mockResponse)
+
+	github, _ := New("Wattpad", "fake-access-token")
+	url, _ := github.GetTravisCIBuildURLForRef("highlander", "2c76895cdb9f3ff5100ecf93a7a6c6747aaeda8c")
+
+	assert.Equal(t, "https://travis-ci.com/Wattpad/highlander/builds/115827260", url)
+}
+
+func TestGetTravisCIBuildURLForRef_Empty(t *testing.T) {
+	defer gock.Off()
+
+	mockResponse, _ := ioutil.ReadFile("testdata/check_runs_response_empty.json")
+	gock.New("https://api.github.com").
+		Get("/repos/Wattpad/highlander/commits/2c76895cdb9f3ff5100ecf93a7a6c6747aaeda8c/check-runs").
+		Reply(200).
+		JSON(mockResponse)
+
+	github, _ := New("Wattpad", "fake-access-token")
+	_, err := github.GetTravisCIBuildURLForRef("highlander", "2c76895cdb9f3ff5100ecf93a7a6c6747aaeda8c")
+
+	assert.Equal(t, "could not find a TravisCI build for specified ref", err.Error())
```

Comment:
Using [`assert.EqualError`](https://godoc.org/github.com/stretchr/testify/assert#EqualError) would be slightly cleaner, but I don't like that this test will fail if the message of the error returned by `GetTravisCIBuildURLForRef` ever changes. 

Tests should verify a behaviour of the implementation (return an error when the response contains no check-runs), not the implementation details of that behaviour (the specific string in the error message)

My suggestion is to define a `var errTravisCIBuildNotFound`, then use that var in both the test cases and the code being tested.